File: assignment_2/View/RunExample.java
package View;

import Controller.Controller;

public class RunExample extends Command {
    private Controller controller;

    RunExample(String key, String description, Controller ctrl) {
        super(key, description);
        this.controller = ctrl;
    }

    @Override
    public void execute() {
        int prg_num = 1;
        while(!this.controller.isEmpty()) {
            System.out.println("PROGRAM NUMBER: " + Integer.toString(prg_num));
            this.controller.allStep();
            prg_num++;
        }
    }
}

File: Toy_Language_Interpreter_C/View/RunExample.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace a7
{
    class RunExample : Command
    {
        private Controller controller;

        public RunExample(String key, String description, Controller ctrl) : base(key, description)
        {
            this.controller = ctrl;
        }

        public override void Execute()
        {
            int prg_num = 1;
            while(!this.controller.IsEmpty())
            {
                Console.Write("PROGRAM NUMBER: " + prg_num.ToString() + "\n");
                this.controller.AllStep();
                prg_num++;
            }
        }
    }
}
